lib/util/Components.js

```diff
@@ -63,14 +63,50 @@ Components.prototype.get = function(node) {
  * @param {Object} props Additional properties to add to the component.
  */
 Components.prototype.set = function(node, props) {
+  const popsAreEquivalent = (propA, propB) => {
```

Comment:
s/pops/props

can this function be defined at file level, so it doesn't need to be recreated every time `set` is run?